src/sdk/pynni/nni/counter.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import torch
+import torch.nn as nn
+from torch.nn.modules.conv import _ConvNd
+from torch.nn.modules.batchnorm import _BatchNorm
+from nni.compression.torch.compressor import PrunerModuleWrapper
+import logging
+from thop import profile
+
+_logger = logging.getLogger(__name__)
+
+def count_flops_params(model: nn.Module, input_size=None, verbose=False):
+    """
+    Count FLOPs and Params of the given model. This function would identify the mask on the module and take the pruend shape into consideration. Note that, we only identify the output shape of the module, so the calculated FLOPs will be larger than real number.
+
+    Parameters
+    ---------
+    model : nn.Module
+        target model.
+    input_size: list, tuple
+        the input shape of data
+
+    Returns
+    -------
+    flops: float
+        total flops of the model
+    params:
+        total params of the model
+    """
+
+    assert input_size is not None
+
+    device = next(model.parameters()).device
+    inputs = torch.randn(input_size).to(device)
+
+    m_list = list(model.modules())
+    hook_module_list = []
+    for idx, m in enumerate(m_list):
+        weight_mask = None
+        m_type = type(m)
+        if m_type in custom_ops:
+            if isinstance(m_list[idx-1], PrunerModuleWrapper):
```

Comment:
What if idx == 0? What if PrunerModuleWrapper has multiple children?

Reply:
Thanks~ `PrunerModuleWrapper` is our customize wrapper and it directly wraps Conv or Linear module, so it won't have multiple children.